app/Main.hs

```diff
@@ -3,32 +3,40 @@
 
 import System.Environment (getArgs)
 import System.Exit (exitFailure)
-
 import System.IO (hPutStrLn, stderr)
 
+import Control.Exception.Safe
+import Control.Applicative (optional)
 import Data.Semigroup ((<>))
 import Data.Text (unpack)
 import qualified Options.Applicative as Opt
 import Options.Applicative ((<**>), many)
-
-import Control.Exception.Safe
 import PyF
 
 import Krank
 import Krank.Formatter
 
 data KrankOpts = KrankOpts {
-  codeFilePaths :: [FilePath]
+  codeFilePaths :: [FilePath],
+  githubKey :: Maybe String
```

Comment:
Could you wrap `githubKey` in a `newtype`?

Reply:
Following our offline discussion, I did a `newtype` wrap to have a clean function signature 
--> is that what you wanted?

Actually, why `newtype`? My (maybe wrong understanding) is that to do that `data` and `newtype` will do the same but `data` is preferred and `newtype` is rather used when we specifically want a "redefinition" that will implement differently some typeclasses. Am I missing something?

Reply:
Thank you for the change. Actually I was expectding a `Maybe GithubKey` with `GithubKey` just a `newtype` over `String`, but I have no argument for or against your solution.

`newtype` versus `data` is really simple. `newtype` comes for free, no overhead and the same behavior of the wrapped type. Especially regarding strictness:

The ctor for a `data` can be undefined:

```haskell
Prelude> data Poulet = Poulet Int
Prelude> foo (Poulet _) = "Ypipp"

Prelude> foo undefined
"*** Exception: Prelude.undefined
CallStack (from HasCallStack):
  error, called at libraries/base/GHC/Err.hs:78:14 in base:GHC.Err
  undefined, called at <interactive>:3:5 in interactive:Ghci3
Prelude> foo (Poulet undefined)
"Ypipp"
```

In this case, Haskell will first evaluate the first "thunk" to see if there is a constructor inside it, and fail on the `undefined`.

On the other hand:

```haskell
Prelude> newtype Poulet = Poulet Int
Prelude> foo (Poulet _) = "Ypipp"


Prelude> foo undefined
"Ypipp"
```

It behaves exactly as an Int, without wrapping constructor.


As for the reasons to prefer one or the other in this context, there is no reason, we don't care about performance neither strictness properties. That's just an habit I have.